src/iroh.rs

```diff
@@ -474,159 +422,281 @@ impl VeilidIrohBlobs {
-        let collection_data = self.read_bytes(collection_hash).await?;
-        let mut collection: Collection = from_slice(&collection_data)?;
     
-        // Add or update the file in the collection
-        collection.files.insert(path, hash);
     
-        // Serialize the updated collection to CBOR
+    pub async fn set_file(&self, collection_name: &String, path: &String, file_hash: &Hash) -> Result<Hash> {
+         // Retrieve the current collection hash from the tag
+        println!("Attempting to retrieve tag for collection: {}", collection_name);
+
+        // Retrieve the collection hash using `collection_hash` (use this to get the collection)
+        let collection_hash = match self.collection_hash(&collection_name).await {
+            Ok(hash) => hash,
+            Err(e) => {
+                println!("Error retrieving tag for collection: {:?}", e);
+                return Err(anyhow!("Tag retrieval failed for collection: {:?}", e));
+            }
+        };
+
+        println!("Successfully retrieved tag for collection: {}", collection_name);
+
+        // Fetch collection data by its hash
+        let collection_entry = self.store.get(&collection_hash).await?;
+        let mut collection: HashMap<String, Hash> = if let Some(entry) = collection_entry {
+            let mut reader = entry.data_reader();
+            let collection_data: Bytes = reader.read_to_end().await?;
+            match from_slice(&collection_data) {
+                Ok(deserialized_collection) => deserialized_collection,
+                Err(err) => {
+                    println!("Error deserializing collection: {:?}", err);
+                    return Err(anyhow!("Failed to deserialize collection: {:?}", err));
+                }
+            }
+        } else {
+            HashMap::new()
+        };
+
+        // Add or update the file in the collection (HashMap)
+        collection.insert(path.clone(), file_hash.clone());
+
+        // Serialize the updated HashMap to CBOR
         let cbor_data = to_vec(&collection)?;
-        
-            
-        // Write the CBOR data to a temporary file
-        let temp_path = self.base_dir.join(format!("{}_updated.cbor", collection_name));
-        std::fs::write(&temp_path, &cbor_data)?;
 
-        // Convert the path to an absolute path
+        // Upload the updated collection data to the store
+        let temp_path = self.base_dir.join("updated_collection.cbor");
+        std::fs::write(&temp_path, &cbor_data)?;
         let absolute_temp_path = std::fs::canonicalize(&temp_path)?;
-
-        // Upload the updated collection
         let new_collection_hash = self.upload_from_path(absolute_temp_path).await?;
+
+        // Log: Print the new collection hash
+        println!("New collection hash: {:?}", new_collection_hash);
+
+        // Store the new collection hash with the tag
+        self.store_tag(&collection_name, &new_collection_hash).await?;
```

Comment:
We should encapsulate the logic of encoding/hashing/tagging into a function

e.g. `let hash = self.update_collection(name, collection).await?`

Reply:
b3fbc4ddf3cab612a93f5f9eef78632ab131cd99